app.js

```diff
@@ -1,20 +1,34 @@
 const express = require("express");
 const app = express();
-
 const { getTopics } = require("./controllers/topics.controllers");
 const getEndPoints = require("./controllers/api.controllers");
+const getArticleId = require("./controllers/articles.controllers");
 
 app.get("/api/topics", getTopics);
+app.get("/api", getEndPoints);
+app.get("/api/articles/:article_id", getArticleId);
 
+app.use((err, req, res, next) => {
+  if (err.status && err.msg) {
+    res.status(err.status).send(err.msg);
+  } else next(err);
+});
 
-app.get("/api", getEndPoints)
+app.use((err, req, res, next) => {
+  if (err.code === "22P02") {
+    res.status(400).send({ msg: "Invalid Input" });
+  } else next(err);
+});
 
 app.use((err, req, res, next) => {
-  res.status(500).send("Internal server error");
+  if (err.status===400 && err.msg==="Bad Request") {
+    res.status(400).send("Bad Request");
+  } else next(err);
 });
 
-app.use((request, response) => {
-  response.status(404).send("Not found");
+app.use((err, req, res, next) => {
+  console.log(err);
+  res.status(500).send("Internal server error");
```

Comment:
Above (line 19) you have sent your error message as an object with a key of msg which is great however you have not been consitent with your other error handling and this will cause problems later.